sql/core/src/main/scala/org/apache/spark/sql/execution/ExistingRDD.scala

```diff
@@ -116,10 +116,17 @@ case class LogicalRDD(
       case e: Attribute => rewrite.getOrElse(e, e)
     }.asInstanceOf[SortOrder])
 
+    val rewrittenOriginLogicalPlan = originLogicalPlan.map { plan =>
+      val projectList = output.map { attr =>
+        Alias(attr, attr.name)(exprId = rewrite.getOrElse(attr, attr).exprId)
```

Comment:
As `rewrite` is a map for all output. We already can get `rewrite(attr)`. `getOrElse` is redundant.

Reply:
This is more about the sake of defensive programming - if there is a bug which makes the two set of columns be out of sync, we just allow them to be out of sync in future instead of failing the query, given that the impact of two set of columns be out of sync is not that quite serious, e.g. column stat won't be available.

In opposite way, I'm also in favor of fail-fast, setting the precondition that "two set of columns should be in sync", and assert the precondition on initialization of the class. After that we can safely assume that precondition is respected, and then it'd be safe to just use `rewrite(attr)` here.

I'm fine either way. WDYT? cc. @cloud-fan as well.

Reply:
Okay for me. Just a nit comment.

Reply:
No, not really. My bad you're right. It only looks into the output of LogicalRDD. Let me reflect the change.